File: Python3/594. Longest Harmonious Subsequence.py
# 404ms, 26.47%; 16.1MB, 7.69%
from collections import Counter


class Solution:
    def findLHS(self, nums: list) -> int:
        n = sorted(Counter(nums).items())
        res = 0
        for i in range(1, len(n)):
            if n[i - 1][0] + 1 < n[i][0]: continue
            res = max(res, n[i - 1][1] + n[i][1])
        return res

File: 401 - 600/594 Longest Harmonious Subsequence.cc
// 104ms, 54.50%; 20.6MB, 33.33%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findLHS(const vector<int> &nums) {
        if (nums.empty()) return 0;
        map<int, int> tmp;
        for (auto n : nums) ++tmp[n];
        int res = 0;
        for (auto it = next(tmp.begin()); it != tmp.end(); ++it) {
            if (prev(it)->first + 1 < it->first) continue;
            res = max(res, prev(it)->second + it->second);
        }
        return res;
    }
};
